ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -579,19 +589,42 @@ applyPatchCmmParseNoImplicitPrelude ghcFlavor =
 -- function 'appyPatchHadrianStackYaml' guarantees it is available
 -- if its needed.
 
--- Patch Hadrian's Cabal.
+data HadrianStackYaml =
+  HadrianStackYaml {
+      extraDeps :: [Y.Value]
+    , ghcOptions :: HMS.HashMap T.Text T.Text
+    , otherFields :: Y.Object
+  } deriving (Eq, Show)
```

Comment:
Can't we just do the modifications on the YAML value without this type and its instances? That should be significantly shorter and probably not harder to understand.

Reply:
I'm sorry, you'll have to demonstrate the technique. I don't know how to do that.

Reply:
By the way, what would be the goal? Is there something wrong with this program?

Reply:
parse into `Object` and do a key lookup on the hashmap instead of parsing into a custom type.

Reply:
Ah right, I get you. Well, the more verbose version exists. Is there real value in unwinding it in favor of doing that?

Reply:
It is of course a short distance from here to there compared to from where we were.

Reply:
it's done